services/static-webserver/client/source/class/osparc/ui/list/CollaboratorListItem.js

```diff
@@ -149,62 +165,79 @@ qx.Class.define("osparc.ui.list.CollaboratorListItem", {
       });
 
       const accessRights = this.getAccessRights();
+      let currentRole = this.self().ROLES[1];
+      if (this.self().canDelete(accessRights)) {
+        currentRole = this.self().ROLES[3];
+      } else if (this.self().canWrite(accessRights)) {
+        currentRole = this.self().ROLES[2];
+      }
 
-      const makeOwnerButton = new qx.ui.menu.Button(this.tr("Make Owner"));
-      makeOwnerButton.addListener("execute", () => {
-        this.fireDataEvent("makeOwner", {
-          gid: this.getKey(),
-          name: this.getTitle()
-        });
-      });
-      const makeCollabButton = new qx.ui.menu.Button(this.tr("Make Collaborator"));
-      makeCollabButton.addListener("execute", () => {
-        this.fireDataEvent("makeCollaborator", {
-          gid: this.getKey(),
-          name: this.getTitle()
-        });
-      });
-      const makeViewerButton = new qx.ui.menu.Button(this.tr("Make Viewer"));
-      makeViewerButton.addListener("execute", () => {
-        this.fireDataEvent("makeViewer", {
-          gid: this.getKey(),
-          name: this.getTitle()
-        });
-      });
+      // promote/demote actions
+      switch (currentRole.id) {
+        case "read": {
+          const promoteButton = new qx.ui.menu.Button(this.tr("Promote to ") + this.self().ROLES[2].label);
```

Comment:
Just wondering if it would be possible to have an arrow pointing up in the ":arrow_up: Promote to "

Reply:
Will check 👍